.github/workflows/validate-nginx-config.yml

```diff
@@ -0,0 +1,77 @@
+name: Validate Nginx Configuration
+on:
+  workflow_call:
+    inputs:
+      custom_headers:
+        description: 'Custom headers to be passed to the Nginx configuration via environment variables.'
+        required: false
+        type: string
+      port:
+        default: 8080
+        description: 'Nginx container port number.'
+        required: false
+        type: number
+      templatedir: # Ref: https://github.com/nginxinc/docker-nginx-unprivileged/blob/main/entrypoint/20-envsubst-on-templates.sh#L32
+        description: 'Nginx configuration template directory. Usually have files with .template extension.'
+        required: true
+        type: string
+      verbose:
+        default: true
+        description: 'Toggle verbose output. By setting this to true, the job will print out the nginx version, nginx config files, and generated config files.'
+        required: false
+        type: boolean
+      workdir:
+        default: 'web'
+        description: 'Default working directory to run the job.'
+        required: true
+        type: string  
+
+jobs:
+  validate-nginx-config:
+    name: validate-nginx-config
+    runs-on: ubuntu-latest
+    env:
+      PORT: ${{ inputs.port }}
+      NGINX_ENVSUBST_TEMPLATE_DIR: ${{ inputs.templatedir}}
+    defaults:
+      run:
+        working-directory: ${{ inputs.workdir }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: check Nginx version
+        run: sudo nginx -v
+        if: ${{ inputs.verbose }}
+      - name: Parse custom headers and export as environment variables
```

Comment:
<img width="1725" alt="image" src="https://github.com/user-attachments/assets/9f7155d2-a68b-4eaa-94b5-1679d458bd15" />

@joshuagibeonm we can dynamically put map of headers to be exported and templatized later via this step

See: https://github.com/reearth/reearth-cms/actions/runs/13514207554/job/37759879232

Reply:
I will continue this discussion on PR comment